gutenTAG/base_oscillations/interface.py

```diff
@@ -48,8 +53,8 @@ def _generate_trend(self, ctx: BOGenerationContext) -> np.ndarray:
                 trend_series = self.trend.timeseries
         return trend_series
 
-    def generate_timeseries_and_variations(self, ctx: BOGenerationContext, **kwargs):
-        self.timeseries = self.generate_only_base(ctx, **kwargs)
+    def generate_timeseries_and_variations(self, ctx: BOGenerationContext, semi_supervised: Optional[bool] = None, supervised: Optional[bool] =  None, **kwargs):
+        self.timeseries = self.generate_only_base(ctx, semi_supervised=semi_supervised, supervised=supervised, **kwargs)
```

Comment:
You don't need this change because the parameters are passed over to the function using `kwargs` anyway. In addition, this change would be required in all subclasses!

Reply:
I took this out, but now generate_only_base, doesn't know whether it is called to create a test, train_no_anomaly or train_anomaly base oscillation. How would I access this information to select the correct input?

Reply:
Sry, I wasn't clear enough: I was just talking about this specific function. The implementation of the feature is correct, but specifying the parameter here is not necessary.

So, reverting the changes in [`timeseries.py`](https://github.com/HPI-Information-Systems/gutentag/pull/6/commits/f9d94e601ca6538977bd03a375976952fa611bb0#diff-32c3ad8ed31d7514cb99823efda484e4c48f71ea88d59546090675522be2c899) and in [`consolidator.py`](https://github.com/HPI-Information-Systems/gutentag/pull/6/commits/f9d94e601ca6538977bd03a375976952fa611bb0#diff-df0c0bd3107b5daec3b7bfb635c86a8944b1eae615dc3c17f34a95d3ddd91968) was not necessary.

Reply:
Thanks. It works again after undoing the reversions in `timeseries.py` and `consolidator.py`